src/librustc_typeck/diagnostics.rs

```diff
@@ -1258,18 +1262,53 @@ extern "rust-intrinsic" {
 "##,
 
 E0107: r##"
-This error means that an incorrect number of lifetime parameters were provided
-for a type (like a struct or enum) or trait:
+This error means that an incorrect number of type or lifetime parameters
+were provided:
 
 ```compile_fail,E0107
 struct Foo<'a, 'b>(&'a str, &'b str);
 enum Bar { A, B, C }
 
 struct Baz<'a> {
-    foo: Foo<'a>, // error: expected 2, found 1
-    bar: Bar<'a>, // error: expected 0, found 1
+    foo: Foo<'a>, // error: wrong number of lifetime arguments:
+                  //        expected 2, found 1
+    bar: Bar<'a>, // error: wrong number of lifetime arguments:
+                  //        expected 0, found 1
+}
+```
+
+```compile_fail,E0107
```

Comment:
These examples can be in the same code block. You don't need an example of less than/greater than the expected number for every single position — maybe one example for each position (type, function call, method call).